sql/driver/conn.go

```diff
@@ -56,7 +56,7 @@ func (c *conn) Prepare(query string) (driver.Stmt, error) {
 }
 
 func (c *conn) Begin() (driver.Tx, error) {
-	c.session = []byte{}
+	c.session = nil
```

Comment:
I still don't understand why you're clearing the session state in transaction begin.